staging/src/k8s.io/kubectl/pkg/describe/describe.go

```diff
@@ -4357,7 +4357,8 @@ func describeNetworkPolicy(networkPolicy *networkingv1.NetworkPolicy) (string, e
 		w := NewPrefixWriter(out)
 		w.Write(LEVEL_0, "Name:\t%s\n", networkPolicy.Name)
 		w.Write(LEVEL_0, "Namespace:\t%s\n", networkPolicy.Namespace)
-		w.Write(LEVEL_0, "Created on:\t%s\n", networkPolicy.CreationTimestamp)
+		// Use a pointer to the CreationTimestamp because *v1.Time is a Stringer.
+		w.Write(LEVEL_0, "Created on:\t%s\n", &networkPolicy.CreationTimestamp)
```

Comment:
Why did this have to change?  What did it output before?

Reply:
This was only changed because the test expected a certain format. Specifically this line: https://github.com/kubernetes/kubernetes/blob/c6935ad50e0809c1dc2d3ac9000abda0ecb49bcc/staging/src/k8s.io/kubectl/pkg/describe/describe_test.go#L4565

If the `&` was removed here, then the output would be `{2017-06-04 21:45:56 -0700 PDT}` (which curly braces).

Reply:
> If the `&` was removed here, then the output would be `{2017-06-04 21:45:56 -0700 PDT}` (with curly braces).

That is so weird.  This is almost like a golang riddle.  I have expanded your example with an OldTime for comparison and I'm still trying to wrap my head around the differences. https://play.golang.org/p/0BtiqauVhtm

I don't really care for an outcome where .String serialization in normal use-cases changes.

Reply:
> I don't really care for an outcome where .String serialization in normal use-cases changes

Playing further with the playground, it looks like if we had the method receiver be the struct, the change in serialization goes away: https://play.golang.org/p/PTBw2MI1Kwe

So `(t time) String`

Reply:
https://play.golang.org/p/-5bHcdAJx2a shows that %s/%v/%#v behavior of putting the direct String() implementation as a struct and pointer receiver.

It also shows that the behavior of a struct receiver is identical to a custom type that implements String() without any added embedded complications.

My inclination is to go with `func (t Time) String() string` and expect debuggers to be smart enough to avoid dereferencing nil pointers to types implementing String() on their struct receiver.

Reply:
Adding a String method on the struct doesn't accomplish anything except adding another method. The functionality will be exactly the same as it currently is and the issue I am trying to fix will remain.

It seems that the real answer is that there is currently no acceptable solution to the problem I am having. This is completely fine with me; it was always one of the possible outcomes.

If both of you agree, then this PR can be closed/rejected and the corresponding issue can be closed as "won't fix."